File: Semestrul4/AOP/Proiect/proiect/repository/RepositoryException.java
package proiect.repository;

// Represents a specialized exception class.
public class RepositoryException extends Exception{
    // Creates a new RepositoryException instance having the specified message.
    public RepositoryException(String message){
        super(message);
    }
    
    private static final long serialVersionUID = 1L;
}

File: Semestrul3/BD/L4/Repository/RepositoryException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BDLab4.Repository
{
    /// <summary>
    /// Represents a specialized exception for repositories.
    /// </summary>
    class RepositoryException : Exception
    {
        /// <summary>
        /// Creates a new instance of RepositoryException
        /// </summary>
        /// <param name="message">is a message that describes the context in which the exception occurred.</param>
        public RepositoryException(string message)
            : base(message)
        {
        }
    }
}
